### PR TITLE
[Refactor](nereids)Refactor Literal to an inheritance hierarchy

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: no inssue

## Problem Summary:

Use inheritance hierarchy instead of combination to make the framework more clear

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BooleanLiteral.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class BooleanLiteral extends Literal {
+
+    public static final BooleanLiteral TRUE_LITERAL = new BooleanLiteral(true);
+    public static final BooleanLiteral FALSE_LITERAL = new BooleanLiteral(false);
+
+    private final boolean value;
+
+    public BooleanLiteral(boolean value) {
+        super(IntegerType.INSTANCE);
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BooleanLiteral.java:34]: Do we need to introduce `BooleanType.INSTANCE`?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BooleanLiteral.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.doris.nereids.types.BooleanType;
+
+
+/**
+ * Represents Boolean literal
+ */
+public class BooleanLiteral extends Literal {
+
+    public static final BooleanLiteral TRUE_LITERAL = new BooleanLiteral(true);
+    public static final BooleanLiteral FALSE_LITERAL = new BooleanLiteral(false);
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BooleanLiteral.java:29]: How about simplifying `TRUE_LITERAL` and `FALSE_LITERAL` to `TRUE` and `FALSE`.

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BooleanLiteral.java]: done

**Review by adonis0147 — DISMISSED** [review]
LGTM

**Review by kpfly — DISMISSED** [review]
LGTM

**Review by morrySnow — DISMISSED** [review]

**Comment by 924060929** [comment]
missing visit/accept function?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Literal.java [hunk 3]
```diff
@@ -85,12 +67,12 @@ public DataType getDataType() throws UnboundException {
 
     @Override
     public String toSql() {
-        return value.toString();
+        return String.valueOf(getValue());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Literal.java:70]: NullLiteral and BooleanLiteral should return upper case in toSql and toString

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Literal.java]: done

**Review by morrySnow — APPROVED** [review]

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**